static/app/components/replays/breadcrumbs/replayTimelineEvents.tsx

```diff
@@ -57,86 +56,127 @@ const EventColumn = styled(Timeline.Col)<{column: number}>`
   }
 `;
 
-function getCrumbDetail(crumb: Crumb) {
-  switch (crumb.type) {
-    case BreadcrumbType.ERROR:
-      return `${crumb.data?.label}: ${crumb.message}`;
-    default:
-      return getDescription(crumb);
-  }
-}
-
-function sortByMostInterestingType(a: Crumb, b: Crumb) {
-  const rank = {
-    [BreadcrumbType.USER]: 0,
-    [BreadcrumbType.UI]: 0,
-    [BreadcrumbType.NAVIGATION]: 1,
-    [BreadcrumbType.ERROR]: 2,
-  };
-
-  return rank[a.type] - rank[b.type];
-}
-
-function Event({crumbs}: {crumbs: Crumb[]; className?: string}) {
-  const title = (
-    <HoverList>
-      {crumbs.map(crumb => (
-        <HoverListItem key={crumb.id}>
-          <Type type={crumb.type} color={crumb.color} description={crumb.description} />
-          <small>{getCrumbDetail(crumb) || getTitle(crumb)}</small>
-        </HoverListItem>
-      ))}
-    </HoverList>
-  );
-
-  const mostInteresting = crumbs.reduce((best, crumb) =>
-    sortByMostInterestingType(best, crumb) >= 1 ? best : crumb
+function Event({
+  crumbs,
+  startTimestamp,
+}: {
+  crumbs: Crumb[];
+  startTimestamp: number;
+  className?: string;
+}) {
+  const {setCurrentTime} = useReplayContext();
+
+  const handleClick = useCallback(
+    (crumb: Crumb) => {
+      crumb.timestamp !== undefined
+        ? setCurrentTime(relativeTimeInMs(crumb.timestamp, startTimestamp))
+        : null;
+    },
+    [setCurrentTime, startTimestamp]
   );
 
-  const icon = crumbs.length === 1 ? <Icon type={mostInteresting.type} /> : crumbs.length;
+  const title = crumbs.map(crumb => (
+    <BreadcrumbItem
+      key={crumb.id}
+      crumb={crumb}
+      startTimestamp={startTimestamp}
+      isHovered={false}
+      isSelected={false}
+      onClick={handleClick}
+    />
+  ));
+
+  const overlayStyle = css`
+    /* We make sure to override existing styles */
+    padding: ${space(0.5)} !important;
+    max-width: 291px !important;
+    width: 291px;
+
+    @media screen and (max-width: ${theme.breakpoints.small}) {
+      max-width: 220px !important;
+    }
+  `;
+
+  // If we have more than 3 events we want to make sure of showing all the different colors that we have
+  const colors = [...new Set(crumbs.map(crumb => crumb.color))];
+
+  // We just need to stack up to 3 times
+  const totalStackNumber = crumbs.length > 3 ? 3 : crumbs.length;
 
   return (
     <IconPosition>
-      <Tooltip key={mostInteresting.id} title={title} containerDisplayMode="block">
-        <IconNode color={mostInteresting.color}>{icon}</IconNode>
-      </Tooltip>
+      <IconNodeTooltip title={title} overlayStyle={overlayStyle} isHoverable>
+        {Array.from({...crumbs, length: totalStackNumber}).map((crumb, index) => (
```

Comment:
Re-defining `length` here seems a bit odd - I had to re-read this a few times to understand what you're trying to do. Instead I would use `slice()` to get the items you need from the array.

`crumbs.slice(0, totalStackNumber).map(...)`